test/service/test_request_validation.py

```diff
@@ -121,7 +120,10 @@ def test_bad_sort_facet_and_filter_facet_of_sample(self):
         }
         response = requests.get(url, params=params)
         self.assertEqual(400, response.status_code, response.json())
-        self.assertIn(response.json(), [self.facet_message, self.facet_enumeration_message()])
+        try:
+            self.assertEqual(self.facet_message, response.json())
+        except AssertionError:
+            self.assertIn(self.facet_enum_message, response.json()['Message'])
```

Comment:
I had to use try: except because if there was a failing in one of the tests, the OR statement does not get evaluated, the test failure is raised.

```
Traceback (most recent call last):
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pycharm/teamcity/diff_tools.py", line 32, in _patched_equals
    old(self, first, second, msg)
  File "/Users/amar/.pyenv/versions/3.8.2/lib/python3.8/unittest/case.py", line 912, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/Users/amar/.pyenv/versions/3.8.2/lib/python3.8/unittest/case.py", line 1211, in assertDictEqual
    self.fail(self._formatMessage(msg, standardMsg))
  File "/Users/amar/.pyenv/versions/3.8.2/lib/python3.8/unittest/case.py", line 753, in fail
    raise self.failureException(msg)
AssertionError: {'Cod[47 chars]ror: Unknown facet 'bad-facet'"} != {'Cod[47 chars]ror: Invalid input for 'sort'. Must be one of [794 chars]']."}
Diff is 1201 characters long. Set self.maxDiff to None to see it.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/amar/.pyenv/versions/3.8.2/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/Users/amar/.pyenv/versions/3.8.2/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/Users/amar/.pyenv/versions/3.8.2/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/Users/amar/dev/hca/azul/test/service/test_request_validation.py", line 124, in test_bad_sort_facet_and_filter_facet_of_sample
    (self.assertEqual(self.facet_message, response.json()) or
```

Reply:
Your reasoning makes sense. I'm wondering now why the assert error is being raised in the first place. Assertions being raised indicate a problem with the code, so catching one in a test is a bad smell. But maybe I'm misunderstanding what's going on here.